src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    protected MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketIndex = getHash(key);
+        resize();
+        Node<K, V> newNode = new Node<>(getHash(key), key, value, null);
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = newNode;
+        } else {
+            Node<K, V> currentNode = table[bucketIndex];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = newNode;
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int bucketNumber = getHash(key);
+        Node<K, V> newNode = table[bucketNumber];
```

Comment:
I think you should rename this variable because you get the needed node, not a new node